pkg/addons/manifest.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2020 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	"github.com/kubermatic/kubeone/pkg/state"
+
+	metav1unstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+func getManifests(s *state.State) error {
+	vars := map[string]interface{}{
+		"KubeOne": s.Cluster,
+	}
+	manifests, err := loadAddonsManifests(s.Cluster.Addons.Path, s.Verbose, vars)
```

Comment:
I'm wondering should we have a CLI flag instead of the API structure. For example, something such as `kubeone install config.yaml --addons ./addons`. Maybe it would improve the user experience, as we would avoid situations such as that one. WDYT about it?

Reply:
Hm, and the fact of setting `--addons` by the called can be treated as a signal to enable addons handling, and wise-versa, lack of this flag means — don't do anything about addons.